Source/SaveStates/SaveState.cs

```diff
@@ -148,19 +157,7 @@ public void LoadStateFromFile(int paramSlot)
                     SaveStateData tmpData = JsonUtility.FromJson<SaveStateData>(File.ReadAllText(data.filePath));
                     try
                     {
-                        data.saveStateIdentifier = tmpData.saveStateIdentifier;
-                        data.cameraLockArea = tmpData.cameraLockArea;
-                        data.savedPd = tmpData.savedPd;
-                        data.savedSd = tmpData.savedSd;
-                        data.savePos = tmpData.savePos;
-                        data.saveScene = tmpData.saveScene;
-                        data.lockArea = tmpData.lockArea;
-                        data.isKinematized = tmpData.isKinematized;
-                        if (tmpData.loadedScenes is not null)
```

Comment:
This check if important. It helps older made save state files load with new versions. You should also add a check for the new array you added to the data

Reply:
Check is already done in the ctor (c.f. L55), this is just duplicate code

Reply:
ah ok then nvm